src/fairseq2/generation/text.py

```diff
@@ -200,6 +205,9 @@ def __call__(self, source_text: str) -> Tuple[str, Seq2SeqGeneratorOutput]:
         """
         source_seq = self._source_text_encoder(source_text)
 
+        if self.max_src_len and source_seq.shape[0] > self.max_src_len:
```

Comment:
Although it would be a very edge case; if `max_src_len` is 0, then the first condition `if self.max_src_len` will be false, but maybe the user deliberately wants to trim to zero length (although does not make much sense, still expected since our parameter contract says that this parameter is ignored only if it is `None`). The "right" check would be `if self.max_src_len is not None and ...`

Reply:
If the source is trimmed to 0 elements, the translation is going to fail, because the encoder would consume an empty tensor. Therefore, I don't want to apply truncation to zero, and instead will change the parameter contract.